synthesizer/src/store/helpers/memory/internal/mod.rs

```diff
@@ -162,11 +170,6 @@ impl<
 
         // Remove all operations after the checkpoint.
         atomic_batch.truncate(checkpoint);
-
-        // If the atomic batch is now empty, set the atomic batch flag to `false`.
-        if atomic_batch.is_empty() {
-            self.batch_in_progress.store(false, Ordering::SeqCst);
-        }
```

Comment:
I'm only concerned about this: I realize that this is because a single rewind might not mean that the atomic batch is over; however, if the calls of `atomic_batch_scope` end with `?`, cascading to the outer scopes, this could cause a situation where we end up with `batch_in_progress` set to `true` while being outside of an atomic batch.